### ISSUE TITLE
Bug: duplicate issues reported in 2.5

### ISSUE THREAD

**Issue by digininja** [sullo/nikto#734]
### Expected behavior

Each issue should be reported once.

### Actual behavior
In 2.5, a bunch of duplicate issues are reported.   `Server banner changed from '' to 'Apache'` is reported five times

Compare these two files, the 25 file is from the 2.5.0 branch, the other is from master

[nikto.json.txt](https://github.com/sullo/nikto/files/6846678/nikto.json.txt)

[nikto25.json.txt](https://github.com/sullo/nikto/files/6846679/nikto25.json.txt)

### Steps to reproduce

1. `./nikto.pl -h 192.168.0.52 -Format json -o nikto25.json` with both `nikto-2.5.0` and `master`
2.
3.

### Nikto version

Run:

```
./nikto.pl -Version
```

and paste the output here.

### Further technical info

E.g. you can obtain Nikto debug output by running `-D D` and redirecting to a file.
You may also scrub the output of hostnames and IPs by specifying `-D DS`.

**Comment by sullo** [comment]
What is the server banner before it changes to Apache? The logic assumes it was present in the response, it wasn't reported before, the new banner is not the same old banner, and that it's not that stupid HTTPAPI BS. Unless something else is out of whack, the implication of the message is that it sent a header of `Server: ` which seems weird. Can you share the pre/post Server headers? 

```
            if (   exists $result{'server'}
                && !exists $mark->{'bannerchanged'}
                && ($mark->{'banner'} ne $result{'server'})
		        && ($result{'server'} ne 'Microsoft-HTTPAPI/2.0')
		) {
```

**Comment by digininja** [comment]
I'll dig them out tomorrow. The web server is Apache and isn't configured
to do anything unusual or special as it is just my dev box.

**Comment by digininja** [comment]
I've checked the save files and this is all that is logged for the issue:

```
---------------------------------------------------------------------------
              Information
---------------------------------------------------------------------------
Test ID:    999962
References:    
Message:    Server banner changed from '' to 'Apache'.
Reason:    
---------------------------------------------------------------------------
              Request
---------------------------------------------------------------------------
Not Applicable

---------------------------------------------------------------------------
              Response
---------------------------------------------------------------------------
Not Applicable
---------------------------------------------------------------------------
              Data Objects
---------------------------------------------------------------------------
```

So not much help tracking it down.

In all the other issues that are reported, the server comes back as Apache.

```
robin@flint nikto_dvwa $ grep server: * -a
dvwa.test_80_2021-07-21_003584.txt:server: Apache
dvwa.test_80_2021-07-21_006333.txt:server: Apache
dvwa.test_80_2021-07-21_006530.txt:server: Apache
dvwa.test_80_2021-07-21_006609.txt:server: Apache
dvwa.test_80_2021-07-21_006807.txt:server: Apache
dvwa.test_80_2021-07-21_007224.txt:server: Apache
dvwa.test_80_2021-07-21_740001.txt:server: Apache
dvwa.test_80_2021-07-21_740001.txt:server: Apache
dvwa.test_80_2021-07-21_95.txt:server: Apache
dvwa.test_80_2021-07-21_999103.txt:server: Apache
dvwa.test_80_2021-07-21_999957.txt:server: Apache
```

The JSON reports the banner changing 12 times
[nikto_debug.log](https://github.com/sullo/nikto/files/6854999/nikto_debug.log)


```
 $ cat nikto_dvwa.json | jq .vulnerabilities[].msg | grep banner
"Server banner changed from '' to 'Apache'."
"Server banner changed from '' to 'Apache'."
"Server banner changed from '' to 'Apache'."
"Server banner changed from '' to 'Apache'."
"Server banner changed from '' to 'Apache'."
"Server banner changed from '' to 'Apache'."
"Server banner changed from '' to 'Apache'."
"Server banner changed from '' to 'Apache'."
"Server banner changed from '' to 'Apache'."
"Server banner changed from '' to 'Apache'."
"Server banner changed from '' to 'Apache'."
"Server banner changed from '' to 'Apache'."
 $ cat nikto_dvwa.json | jq .vulnerabilities[].msg | grep banner|wc -l
12
```

I scanned again with debug enabled and all the server responses in there contain Apache as far as I can tell, I've attached it for you to have a look at. It might also be useful for you looking at the slow scan so you can spot which requests are taking the extra time.

**Comment by tautology0** [comment]
Yeah it never returns an empty server - a quick fix would be to put in a test for an empty server header and not count it as a change. Just debugging.

**Comment by digininja** [comment]
If it doesn't return an empty header, then what change is it detecting?

**Comment by tautology0** [comment]
I think this is due to #733

**Comment by tautology0** [comment]
Fixed in e15b3ed8553d0f3de66eef63d1a21f35d8294046

**Comment by digininja** [comment]
I'm just nipping out but will rerun my original test when I get back and
let you know how it goes.
